node/hack/repair_solana/repair.go

```diff
@@ -128,16 +128,24 @@ func main() {
 				break
 			}
 
-			last := txs[len(txs)-1].Signature
-
-			_, lastSeq, err := fetchTxSeq(ctx, sr, last)
-			if err != nil {
-				log.Fatalf("fetch last tx seq: %v", err)
+			var lastSeq, firstSeq uint64
+
+			var last solana.Signature
+			for i := 0; lastSeq == 0; i-- {
```

Comment:
it's not entirely clear to me what this loop has to do with the offset change. Is this just an unrelated improvement to make sure we find the last tx that has a sequence number in it?

Reply:
Yep!